### PR TITLE
[ALLUXIO-2236] Fix a bug related channel close

### PR DESCRIPTION
**Pull request by peisun1115**
A channel can be closed asynchronously in netty. we should avoid doing that.

### CODE REVIEW

File: core/client/src/main/java/alluxio/client/netty/NettyRemoteBlockReader.java [hunk 1]
```diff
@@ -121,7 +121,7 @@ public ByteBuffer readRemoteBlock(InetSocketAddress address, long blockId, long
       }
       throw new IOException(e);
     } finally {
-      if (channel != null && listener != null) {
+      if (channel != null && listener != null && channel.isActive()) {
```
Comment (ifcharming) [inline core/client/src/main/java/alluxio/client/netty/NettyRemoteBlockReader.java]: nit: do you also want to check `channel.pipeline().get(ClientHandler.class)` is not-null ? Is it possible in some corner cases, when the channel is still active, but `ClientHandler` is somehow removed from the pipleline?

Comment (peisun1115) [inline core/client/src/main/java/alluxio/client/netty/NettyRemoteBlockReader.java]: i can throw an exception (NullPointerException) explicitly. Is that what you are suggesting?. It should not be null in the current code. If it happens to null, something is working unexpectedly.

Comment (ifcharming) [inline core/client/src/main/java/alluxio/client/netty/NettyRemoteBlockReader.java]: I'd prefer not to crash the production server if some bug leads to this case (e.g. `NettyUnderFileSystemFileWriter` is invoked on the worker side). Does it make sense to print an error message instead in such case?

Comment (peisun1115) [inline core/client/src/main/java/alluxio/client/netty/NettyRemoteBlockReader.java]: Discussed in person. In this case, the server should be crashed. I don't know what is the convention in Alluxio codebase. if we prefer explicit precondtion check. i can do that.

**Comment by peisun1115** [comment]
Jenkins, test this please

**Comment by peisun1115** [comment]
FYI, tests passes locally.  I will wait for jenkins

**Pull request merged**